src/grammar.pest

```diff
@@ -18,8 +18,9 @@ char = {
 }
 
 union =  { "[" ~ unionElement ~ ("," ~ unionElement)* ~ "]" }
-unionElement = _{ unionChild }
+unionElement = _{ unionChild | unionArrayIndex }
 unionChild = { doubleQuotedString | singleQuotedString }
+unionArrayIndex = { "-" ? ~ ASCII_DIGIT+ }
```

Comment:
"Note: the syntax does not allow integers with leading zeros such as 01 and -01."